lib/string_decoder.js

```diff
@@ -130,35 +150,43 @@ StringDecoder.prototype.write = function(buffer) {
 // length that character, and sets this.charReceived to the number of bytes
 // that are available for this character.
 StringDecoder.prototype.detectIncompleteChar = function(buffer) {
+  var buflen = buffer.length;
   // determine how many bytes we have to check at the end of this buffer
-  var i = (buffer.length >= 3) ? 3 : buffer.length;
+  var i = (buflen >= 3) ? 3 : buflen;
```

Comment:
You mean for the `i = 3` case? I didn't try that since I figured people would frown about code duplication between the `i = 3` and `i = buflen` cases. I also did not see what kind performance difference it would make though.